CompilerSource/backend/GameData.cpp

```diff
@@ -369,6 +598,40 @@ ImageData::ImageData(const Image &img):
     pixels(img.data, img.data + img.dataSize){}
 ImageData::ImageData(int w, int h, const uint8_t *data, size_t size):
     width(w), height(h), pixels(data, data + size) {}
+ImageData::ImageData(const std::string& filePath) {
+  unsigned error;
+  unsigned char* image;
+  unsigned pngwidth, pngheight;
+
+  error = lodepng_decode32_file(&image, &pngwidth, &pngheight, filePath.c_str());
+  if (error)
+  {
+    printf("error %u: %s\n", error, lodepng_error_text(error));
```

Comment:
The reason not to do this sort of thing (reading a file) in a constructor is precisely because you can't report this error any better way. This should probably result in a compilation failure, but as it stands, it's a single remark printed to stdout.

Reply:
What do you propose to do? Have the compiler components load the binary datas on demand only if it's not already loaded?

Reply:
Wherever this constructor is used, call a helper function to load the image data, then pass that to this constructor. If loading the image data fails, report through the `user` stream (assuming emake supports that) and abort.

Reply:
Alright: 8f1b70790e33e057859be38918b67c9d6a731d2c

I gave it a do-over structured around using helper methods to load in the binary datas. This way a specific error code can be returned and the process aborted early. You'll need to give that a look over.